
#The trigger 'type' field should have the exact name that each trigger class maps as Trigger::name

/robot/battery_state:
  msg_type: sensor_msgs/msg/BatteryState
  triggers:
    - type: BatteryStateTrigger
      persistance_duration: 3.0
      use_msg_stamp: false

/robot/pose:
  msg_type: geometry_msgs/msg/PoseStamped
  triggers:
    - type: ZoneTriggerWithPoseStamped
      persistance_duration: 10.0
      extra_args:
        trigger_zone: {"min_x": 0.0, "max_x": 10.0, "min_y": 0.0, "max_y": 5.0, "min_z": 0, "max_z": 100.0}

/robot/fix:
  msg_type: sensor_msgs/msg/NavSatFix
  trigger_info:
    - type: ZoneTriggerWithNavSatFix
      persistance_duration: 15.0
      trigger_zone: {"latitude_max": 34.55, "latitude_min": 34.15, "longitude_max": 122.55, "longitude_min": 122.15, "altitude_max": 100.0, "altitude_min": 10.0}
    - type: NavSatInvalidFixTrigger
      persistance_duration: 20.0
      use_msg_stamp: true